codecov_cli/helpers/ci_adapters/github_actions.py

```diff
@@ -67,20 +67,22 @@ def _get_slug(self):
 
     def _get_branch(self):
         branch = os.getenv("GITHUB_HEAD_REF")
-        if branch:
-            return branch
 
-        branch_ref = os.getenv("GITHUB_REF")
+        if branch is None:
+            branch = os.getenv("GITHUB_REF")
 
-        if not branch_ref:
-            return None
+            if branch is None:
+                return None
 
-        match = re.search(r"refs/heads/(.*)", branch_ref)
+            match = re.search(r"refs/heads/(.*)", branch)
+            if match is None:
+                return None
+            branch = match.group(1)
 
-        if match is None:
-            return None
+            # handling for merge queue CI runs
+            if branch and branch.startswith("gh-readonly-queue/"):
+                return branch.split("/")[1]
```

Comment:
for clarity can we do

```suggestion
            if branch:
                branch = branch.split("/")[1] if branch.startswith("gh-readonly-queue/") else branch
                return branch
```